staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/endpointslock.go

```diff
@@ -82,6 +82,9 @@ func (el *EndpointsLock) Update(ler LeaderElectionRecord) error {
 	if err != nil {
 		return err
 	}
+	if el.e.Annotations == nil {
+		el.e.Annotations = make(map[string]string)
+	}
```

Comment:
Can you please add a unit test to test the behavior that may have triggerred panic before?
[As described in https://github.com/kubernetes/kubernetes/issues/84729 ]

Reply:
I sent a PR to this branch: https://github.com/yutedz/kubernetes/pull/1

There's also a potential panic in the lease lock.